### PR TITLE
style: Make anon boxes account for :visited.

### PR DESCRIPTION
**Pull request by emilio**
This should fix one of the test failures of:

  https://bugzilla.mozilla.org/show_bug.cgi?id=616436

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19551)

### CODE REVIEW

File: components/style/stylist.rs [hunk 1]
```diff
@@ -679,37 +679,19 @@ impl Stylist {
         parent: Option<&ComputedValues>,
         cascade_flags: CascadeFlags,
         font_metrics: &FontMetricsProvider,
-        rule_node: &StrongRuleNode
+        rule_node: StrongRuleNode
     ) -> Arc<ComputedValues> {
-        // NOTE(emilio): We skip calculating the proper layout parent style
```
Comment (heycam) [inline components/style/stylist.rs]: I see the comment in `compute_pseudo_element_style_with_inputs` already mentions this, but maybe it's to maintain the concrete example of fieldset?

**Review by heycam — APPROVED** [review]

File: components/style/stylist.rs (hunk 1 above)
Comment (emilio) [reply components/style/stylist.rs]: Sure thing :)

**Pull request merged**